py/server/tests/test_partitioned_table.py

```diff
@@ -128,6 +129,10 @@ def test_transform(self):
             pt = self.partitioned_table.transform(Transformer)
             self.assertIn("f", [col.name for col in pt.constituent_table_columns])
 
+            with shared_lock(self.test_table):
```

Comment:
Why wasn't the auto-locking context good enough for this?

Reply:
Because the test table is not ticking I think.

Reply:
OK, I understand the issue. My code expects you to only add extra dependencies if they are refreshing.

Reply:
Seems like we want a test with a live dependency to make sure that the auto-locking works as expected.